### DISCUSSION TITLE
Remove wildcard routes

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [elimintz/justpy#530] [Ideas]
The current design of justpy sets two routes with a wildcard path:

```python
JpRoute("/{path:path}", func_to_run, name="default")
```
```python
@app.route("{path:path}")
class Homepage(JustpyEndpoint):
    """
    Justpy main page handler
    """
```
The motivation seems to be that any path after http://server:port will lead to serving the justpy homepage. 
There might also be usecases where the path is analyzed and used for some purpose.

currently i only see the ajax case:
```python
 if request["path"] == "/zzz_justpy_ajax":
```

This defeats the RESTFul routing idea that is implemented in starlette and ASGI frameworks in general. There are also
other followup issues such as returning a webpage instead of creating a 404 result code if the path is not handled correctly.
As part of the refactoring according to #478 and #502 I suggest to change the behavior and not use a wildcard route any more.

```python
@app.route("/")
class Homepage(JustpyEndpoint):
    """
    Justpy main page handler
    """
```

**Comment by elimintz** [comment]
There could be an issue with path parameters if we do this:
https://justpy.io/tutorial/request_object/#path-parameters

**Comment by sandeep-gh** [comment]
Is this still an issue? Isn't this solved now that  the code is hooks into starlette routing?
